hydra-node/test/Hydra/Model.hs

```diff
@@ -366,14 +371,23 @@ data Nodes m = Nodes
 --
 -- We could perhaps getaway with it and just have a type based on `IOSim` monad
 -- but this is cumbersome to write.
-newtype RunMonad m a = RunMonad {runMonad :: StateT (Nodes m) m a}
-  deriving newtype (Functor, Applicative, Monad, MonadState (Nodes m))
+newtype RunMonad m a = RunMonad {runMonad :: ReaderT (RunState m) m a}
+  deriving newtype (Functor, Applicative, Monad, MonadReader (RunState m), MonadThrow)
 
 instance MonadTrans RunMonad where
   lift r = RunMonad $
-    StateT $ \s -> do
+    ReaderT $ \_ -> do
       a <- r
-      pure (a, s)
+      pure a
```

Comment:
I think a `lift = RunMonad . lift` would yield the same semantics, no?

Reply:
Definitely, perhaps event a `deriving`, let me try